rulesets/src/collections.ruleset.yml

```diff
@@ -0,0 +1,136 @@
+rules:
+  ##### General #####
+  sps-no-collection-paging-capability:
+    description: Response bodies from collection endpoints must offer paging capability.
+    severity: error
+    given: $.paths[?(!@property.match(/.*\/\{[^}]+\}$/))].get.responses['200'].content.application/json.schema.properties
+    then:
+      - field: "paging"
+        function: truthy
+      - field: "paging.type"
+        function: pattern
+        functionOptions:
+          match: "object"
+
+  ##### Root Element #####
+  sps-collection-missing-results-array:
+    description: Response bodies must have a root element called results and is an array of objects.
+    severity: error
+    given: $.paths[?(!@property.match(/.*\/\{[^}]+\}$/))].get.responses['200'].content.application/json.schema.properties.results
+    then:
+      - field: type
+        function: pattern
+        functionOptions:
+          match: "array"
+      - field: items.type
+        function: pattern
+        functionOptions:
+          match: "object"
+
+  ##### Pagination #####
+  sps-missing-pagination-query-parameters:
+    description: 'Collection GET endpoints must support pagination using query parameters: limit, offset, cursor.'
+    severity: error
+    given: $.paths[?(!@property.match(/.*\/\{[^}]+\}$/))].get
+    then:
+      - field: parameters
+        function: schema
+        functionOptions:
+          schema:
+            type: array
+            items:
+              type: object
+              properties:
+                name:
+                  type: string
+              required:
+                - name
+            minItems: 3
```

Comment:
what are the implications of this? why is a property with "name" required?

Reply:
This is removed from my updates